app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+cache_archive = {}
+
+
+def cache(func: Callable) -> Callable:
+    def wrapper(*args: Any, **kwargs: Any) -> Any:
+        if f"{func} {args} {kwargs}" not in cache_archive.keys():
+            print("Calculating new result")
+            res = func(*args, **kwargs)
```

Comment:
You don’t need this variable res. You return similar data in line 14 without it. And actually, you can return this only one time, now you repeat yourself

Reply:
reworked, tried to make it easier to read